pandas/tests/groupby/aggregate/test_aggregate.py

```diff
@@ -1301,3 +1301,11 @@ def test_group_mean_datetime64_nat(input_data, expected_output):
 
     result = data.groupby([0, 0, 0]).mean()
     tm.assert_series_equal(result, expected)
+
+
+def test_groupby_mean_complex():
```

Comment:
Can we also test other reductions with complex while we're at it? Just would want to make sure that changing the condition in `_get_result_dtype` doesn't cause something else to fail with complex that didn't before

Reply:
among var, median, mean.. only the output of mean changed.
var still returns `TypeError : No matching signature found`, and median still returns absurd results. Although I am little confused how median would be defined for complex numbers (maybe separately calculate mean for real and complex part?)

```python
In [8]: import pandas as pd
   ...: import numpy as np
   ...: 
   ...: data = Series(np.arange(20).reshape(10, 2).dot([1, 2j]))
   ...: funcs=["mean", "median", "var"]
   ...: 
   ...: for x in funcs:
   ...:     print("\n"+x+"\n")
   ...:     try:
   ...:         print(data.groupby(data.index%2).agg(x))
   ...:     except Exception as e:
   ...:         print(e)
   ...: 

mean

0     8.0+18.0j
1    10.0+22.0j
dtype: complex128              #PR

No matching signature found    #master

median

/Volumes/SD/pandas-debnathshoham/pandas/core/groupby/ops.py:192: ComplexWarning: Casting complex values to real discards the imaginary part
  values = ensure_float64(values)
0     8.0
1    10.0
dtype: float64

var

No matching signature found

Reply:
Yeah I guess complex numbers have no ordering, so `min/max` or `median` don't make sense (if they return absurd results, that's another issue since we should probably raise (or just match `numpy` behavior)). `sum` should work though so we should make sure that's tested

Reply:
sure, will add `sum` to this test